pytest_html/plugin.py

```diff
@@ -215,23 +220,55 @@ def pytest_sessionfinish(self, session):
             html.title('Test Report'),
             html.style(raw(style_css)))
 
+        class Outcome:
+
+            def __init__(self, outcome, total=0, label=None,
+                         test_result=None, class_html=None):
+                self.outcome = outcome
+                self.label = label or outcome
+                self.class_html = class_html or outcome
+                self.total = total
+                self.test_result = test_result or outcome
+
+                self.generate_checkbox()
+                self.generate_summary_item()
+
+            def generate_checkbox(self):
+                data_test_result = {"data-test-result":
+                                    self.test_result.lower()}
+                if(self.total > 0):
+                    self.checkbox = html.input(type='checkbox',
+                                               checked='true',
+                                               onChange='filter_table(this)',
+                                               name='filter_checkbox',
+                                               **data_test_result)
+                else:
+                    self.checkbox = html.input(type='checkbox',
+                                               disabled='true')
```

Comment:
Intead of having two different values for `checkbox` you could have one, but make the value of `disabled` dependent on the number of tests with the outcome. For example:

```python
self.checkbox = html.input(
    type='checkbox',
    disabled='true' if self.total > 0 else 'false',
    ...)
```

The remaining attributes will be present for the disabled checkbox, but that won't matter because it's disabled.